Add docs for running a new EN using AWS AMI (#115)
kaiachain/kaia-docs: docs/learn/storage/aws-ami.md

> # Kaia Endpoint Node AMIs

Reviewer: ```suggestion
# Use AWS AMIs
```

Kaia offers an AWS AMI ([Amazon Machine Image](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/AMIs.html)) for Kaia endpoint nodes (EN). These AMIs are pre-installed with both the EN software and the chaindata, enabling users to quickly and conveniently launch a fully operational node. With Kaia's AMI, setting up a new endpoint node on demand becomes seamless, allowing for rapid scaling and simplifying the process of joining the Kaia network.

## Types of EN AMIs

> Kaia provide different types of AMIs, whose chaindata has been synced with different modes.

Reviewer: ```suggestion
Kaia provides different types of AMIs, whose chaindata has been synced with different modes.
```

| **Type** | **Sync Mode** | **AMI Name** |
|----------|---------------|--------------|
| full             | Full mode                                         | `kaia-xxxx-clean-full-en-xxxx` |
| pruning          | Full mode, live pruning enabled                   | `kaia-xxxx-clean-pruning-en-xxxx` |
| state-migrated   | Full mode, state migrated (or state batch pruned) | `kaia-xxxx-clean-en-xxxx`|
| archive          | Archive mode                                      | `kaia-xxxx-clean-archive-en-xxxx` |

> Kaia provide these 4 types of AMIs for both mainnet and kairos.

Reviewer: ```suggestion
Kaia provides these 4 types of AMIs for both mainnet and kairos.
```

> See [State Batch Pruning](../state-pruning/#state-batch-pruning-state-migration) for more details on state migrated chaindata.

Reviewer: ```suggestion
See [State Batch Pruning](../../learn/storage/state-pruning/#state-batch-pruning-state-migration) for more details on state migrated chaindata.
```
This suggested change is supposing that 'aws-ami.md' file is moved under the Node Quick Reference section

> See [Block Syncronization](./block-sync.md) for more details on block sync modes.

Reviewer: ```suggestion
See [Block Synchronization](../../learn/storage/block-sync.md) for more details on block sync modes.
```
The path change is supposing that 'aws-ami.md' file is moved under the Node Quick Reference section

| IP version | Type        | Protocol | Port range     | Source   |
|------------|-------------|----------|----------------|----------|
| IPv4       | Custom TCP  | TCP      | 32323 - 32324  | 0.0.0.0/0|
| IPv4       | Custom UDP  | UDP      | 32323          | 0.0.0.0/0|

## Post-Launch Instance Preparation and Setup

### Warmup Amazon EBS volumes

The Amazon EBS volumes that were created from snapshots (AMI would be one case), the storage blocks must be pulled down from Amazon S3 and written to the volume before they can be accessed. This causes a significant overhead for disk operations the first time each block is accessed. Volume performance is recovered after all blocks have been downloaded and written to the volume. For more details, see [Initialize Amazon EBS volumes](https://docs.aws.amazon.com/ebs/latest/userguide/ebs-initialize.html).

In order to get our volume ready, we can run a task to read all blocks.

```bash
$ sudo yum install -y fio
$ sudo fio --filename=/dev/nvme1n1 --rw=read --bs=128k --iodepth=32 --ioengine=libaio --direct=1 --name=volume-initialize
```

:::note

> This task to warmup Amazon EBS volume would take a couple of days.

Reviewer 2: ```suggestion
This task to warmup Amazon EBS volume would take a long time depending on the data size. Refer to `fio` output for the ETA.
```

:::

### Start `kend` service

In the EC2 instance, Kaia CLI client and the chaindata are installed. Also, `kend`, the script to start/terminate EN, is installed as a service. You can check the `kend` service's status by the following command.

```bash
$ sudo service kend status
```

If the service is not running, try to restart it.

```bash
$ sudo service kend restart
```

If the service restarted and the EN has started successfully, you can check its logs in the path `/var/kend/logs/kend.out`.

```bash
$ tail -f /var/kend/logs/kend.out
```

> Kaia provides a CLI client `ken console`. You can interact with the Kaia node using the `ken console` via multiple endpoints, and one option is to use IPC (inter-process communication). The IPC file `kaia.ipc` is located in the data directory on an EN, in our case `/var/kend/data`. Thus in order to use `ken console`:

Reviewer: The two ipc files 'kaia.ipc' and 'klay.ipc' coexist?

Author: No, my bad. But I'm afraid that there're some `kaia.ipc` in docs..
Anyway, updated. PTAL.

Reviewer: You mean every instance of 'kaia.ipc' in docs needs to be changed to 'klay.ipc' all together?

Author: Yes, currently we use the name of the IPC file as `klay.ipc`.

# Kaia Endpoint Node AMIs

Kaia offers an AWS AMI ([Amazon Machine Image](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/AMIs.html)) for Kaia endpoint nodes (EN). These AMIs are pre-installed with both the EN software and the chaindata, enabling users to quickly and conveniently launch a fully operational node. With Kaia's AMI, setting up a new endpoint node on demand becomes seamless, allowing for rapid scaling and simplifying the process of joining the Kaia network.

## Types of EN AMIs

We provide different types of AMIs for the chaindata synced with different sync modes.

> - plain? TBU e.g., `kaia-xxxx-clean-en-xxxx`

Reviewer 2: Make it a table like in https://docs.kaia.io/misc/operation/chaindata-snapshot/ this page

Author: Done, is that what you asked for?

Reviewer 2: Nice. Thanks!
